src/main/java/com/rentals/service/BookVehicleService.java

```diff
@@ -0,0 +1,77 @@
+package com.rentals.service;
+
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+
+import javax.transaction.Transactional;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Service;
+
+import com.rentals.model.BookVehicle;
+import com.rentals.model.Vehicle;
+import com.rentals.repository.BookVehicleRepository;
+import com.rentals.repository.UserRepository;
+import com.rentals.repository.VehicleModelRepository;
+import com.rentals.repository.VehicleRepository;
+
+@Service
+@Transactional
+public class BookVehicleService {
+
+	private final BookVehicleRepository bookVehicleRepository;
+	private final UserRepository userRepository;
+	private final VehicleRepository vehicleRepository;
+	private final VehicleModelRepository vehicleModelRepository;
+
+	public BookVehicleService(BookVehicleRepository bookingRepository, UserRepository userRepository,
+			VehicleRepository vehicleRepository, VehicleModelRepository vehicleModelRepository) {
+		this.bookVehicleRepository = bookingRepository;
+		this.userRepository = userRepository;
+		this.vehicleRepository = vehicleRepository;
+		this.vehicleModelRepository = vehicleModelRepository;
+	}
+
+	public BookVehicle bookVehicle(BookVehicle bookVehicle) {
```

Comment:
you can have different model for `repository` level and `class` level so that for your logical operations you can use class level model and once you want to save it to db you can parse it to repository model